components/sync15/src/sync.rs

```diff
@@ -36,38 +37,67 @@ pub trait Store {
     /// to handle "backfills" etc
     fn get_collection_request(&self) -> Result<CollectionRequest, failure::Error>;
 
-    fn reset(&self) -> Result<(), failure::Error>;
+    /// Get persisted sync IDs. If they don't match the global state we'll be
+    /// `reset()` with the new IDs.
+    fn get_sync_ids(&self) -> Result<(Option<String>, Option<String>), failure::Error>;
+
+    /// Reset the store without wiping local data, ready for a "first sync".
+    /// The specified guids should be persisted as part of resetting the state
+    /// (ie, after this call succeeds, `get_sync_ids()` should return these values)
+    fn reset(&self, global_id: &str, engine_id: &str) -> Result<(), failure::Error>;
```

Comment:
Given that this gets called with two (weird looking) empty strings often, I think maybe we should add a helper that does that for us if we pass no args.

```
fn full_reset(&self) -> Result<(), failure::Error> { self.reset("", ""); }
```

or something.

Reply:
I ended up making this an `Option<CollSyncIds>`, so now passing `None` seems more reasonable than the old patch, but let me know if you think we still should have a no-arg helper.